hibernate-core/src/main/java/org/hibernate/event/service/internal/EventListenerGroupImpl.java

```diff
@@ -222,6 +222,7 @@ private boolean listenerShouldGetAdded(T listener) {
 							checkAgainstBaseInterface( listener );
 							performInjections( listener );
 							itr.set( listener );
+							listeners = null; //Marks it for refreshing
```

Comment:
not 100% sure but I think the correct thing to do is to remove the original listener from `listeners` .

Reply:
That does not seem the approach used in the class, though

Reply:
forget my comment, I saw how `listerner` it is used